extensions/rest-client/deployment/src/test/java/io/quarkus/restclient/ft/RestClientAnnotationTest.java

```diff
@@ -0,0 +1,97 @@
+package io.quarkus.restclient.ft;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class RestClientAnnotationTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(RestClientAnnotationTest.class, TestEndpoint.class, Client.class));
+
+    static final String URL = "http://localhost:8081";
```

Comment:
We don't want to hardcode this URL, as I know there's work towards supporting random test ports.

There might be a way to get the URL

Reply:
Actually, I wonder if this wouldn't be better to start the test on an entirely separate port?

Then it's actually testing against a port that the app wouldn't be running on normally.

Reply:
Thanks for checking it.
Yes, you are right. The best solution would dynamically get the configuration.

For the configKey I need to create dynamically config property based on the test running URL (I can also delete the src/test/resources/application.properties).
Do we have any example for it?

The baseUri in the annotation is string constant. I have no idea how to fix that.

Reply:
It should be possible to set system variables for the test, to circumvent the need to have them in a static file.

Reply:
I removed the hardcoded URL from the tests

Reply:
Just as an FYI, the random test port support has landed in #4140